gatsby-node.ts

```diff
@@ -17,19 +20,52 @@ export const onCreateWebpackConfig = ({ actions }) => {
 export const createSchemaCustomization = ({ actions: { createTypes } }) => {
   createTypes(`
     type MetaJson implements Node {
-      geoJson: GeometryJson @link(from: "jsonId", by: "name")
-      staticMap: File @link(from: "jsonId", by: "name")
+      geoJson: GeometryJson! @link(from: "jsonId", by: "jsonId")
+      staticMap: File! @link(from: "jsonId", by: "name")
     }
   `);
 };
 
+export const onPostBootstrap = ({ getNodesByType }) => {
+  const nGeometries = getNodesByType('GeometryJson').length;
+  const nMeta = getNodesByType('MetaJson').length;
+  if (nGeometries !== nMeta) {
+    throw new Error(
+      `The provided number of geometries (${nGeometries}) does no match the number of meta information (${nMeta})`
+    );
+  }
+};
+
 export const onCreateNode = async ({
   node,
   actions: { createNode },
   createNodeId,
   cache,
 }) => {
-  if (node.internal.type === 'GeometryJson') {
+  const jsonValidator = new Validator();
+
+  const jsonSchema = {
+    MetaJson: metaJsonSchema,
+    GeometryJson: geometryJsonSchema,
+  };
+  const nodeType = node.internal.type;
+  if (nodeType === 'MetaJson' || nodeType === 'GeometryJson') {
+    // validate meta data against JSON schema
+    const { id, jsonId, parent, children, internal, ...json } = node;
+    try {
+      jsonValidator.validate({ id: jsonId, ...json }, jsonSchema[nodeType], {
+        throwError: true,
+      });
+    } catch (e) {
+      if (e instanceof ValidationError) {
+        console.error(`${jsonId}:`);
+        console.log(e);
```

Comment:
Same here with the [reporter](https://www.gatsbyjs.com/docs/reference/config-files/node-api-helpers/#reporter)

Reply:
I switched to the report for all error throwing